pkgs/development/node-packages/overrides.nix

```diff
@@ -457,7 +457,7 @@ final: prev: {
         version = workerdVersion;
         src = fetchurl {
           url = "https://registry.npmjs.org/@cloudflare/workerd-linux-64/-/workerd-linux-64-${workerdVersion}.tgz";
-          sha512 = "sha512-E8mj+HPBryKwaJAiNsYzXtVjKCL0KvUBZbtxJxlWM4mLSQhT+uwGT3nydb/hFY59rZnQgZslw0oqEWht5TEYiQ==";
+          sha512 = "sha512-2zDcadR7+Gs9SjcMXmwsMji2Xs+yASGNA2cEHDuFc4NMUup+eL1mkzxc/QzvFjyBck98e92rBjMZt2dVscpGKg==";
```

Comment:
This broke because of the way wrangler is updated. Every time you run the [generate.sh](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/node-packages/node-packages.json) it will update the [lock file](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/node-packages/node-packages.nix) which potentially breaks this package because this hash is not updated.

Therefore I'd prefer a fix where you use `buildNpmPackage` to build this as a standalone package, add yourself as a maintainer for the package, remove it from the [node-packages.json](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/node-packages/node-packages.json) and remove these overrides here. This would be a change very similar to the one I did for the [vue-language-server](https://github.com/NixOS/nixpkgs/pull/319134), so you can take inspiration from there.

Reply:
I will take a look and see what I can do. The general idea makes sense to me, seems like the right long-term answer, but I'm not entirely prepared for reworking the whole package. I'll look at the linked pr again and maybe can work something up.

Reply:
Just let me know if you are struggling at any point :)

Reply:
OK so I got a start on this, here is a pr into this pr. Once I'm done testing I'll merge into here if this is still open, otherwise I'll make a new one.

https://github.com/dezren39/nixpkgs/pull/1/files

My first roadblock is the not all dependencies are on npm. `@cloudflare/cli` isn't an npm package, it's available in the github repo in `packages/cli`.

https://github.com/cloudflare/workers-sdk/tree/main/packages/cli

```
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@cloudflare%2fcli - Not found
npm ERR! 404
npm ERR! 404  '@cloudflare/cli@1.1.1' is not in this registry.
npm ERR! 404
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.
```

I'm not sure how best to resolve this part. I did open an issue with cloudflare/workers-sdk asking if they could add this to npm, but maybe there is some other function or method I could use to resolve this on the nixpkgs side?

https://github.com/cloudflare/workers-sdk/issues/6172

I think the function could resolve this dependency if it was listed using github in the `package.json`, but I'm not entirely clear on that.

I had to add `--legacy-peer-deps` in order to get past an issue where my react version was too new for a dependency, 
```
npm ERR! Found: react@18.3.1
npm ERR! node_modules/react
npm ERR!   dev react@"^18.2.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^15.0.0-0 || ^16.0.0-0 || ^17.0.0-0" from @cloudflare/types@6.29.0
npm ERR! node_modules/@cloudflare/types
npm ERR!   dev @cloudflare/types@"^6.18.4" from the root project
```
I'm not sure if there are negative consequences of adding `--legacy-peer-deps` to the npm install command.

I also added `--save-dev` because all the wrangler examples included that flag. This is another one I'm unsure the consequences of. It doesn't seem to make a difference for my current issue, but it may matter once it's actually building.

Reply:
If I understand the package correctly https://github.com/cloudflare/workers-sdk holds all code to build `wrangler` including https://github.com/cloudflare/workers-sdk/tree/main/packages/cli. So instead of fetching `src` for `buildNpmPackage` from npm you can fetch it from github as described in the `buildNpmPackage` example: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/javascript.section.md#buildnpmpackage-javascript-buildnpmpackage. Btw, this readme is a great read anyways.

Reply:
i'll try that! thank you

Reply:
found there is no package-lock because this project uses pnpm but they also use pnpm workspaces running head first into

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/javascript.section.md#dealing-with-sourceroot-javascript-pnpm-sourceroot